### PR TITLE
feat: [CDS-75118]: Updating VirtualServices and DestinationRule without fabric8

### PR DESCRIPTION
**Pull request by GoelPratyush99**
## Describe your changes
Added workflow to not use Fabric8 when FF: CDS_DISABLE_FABRIC8_NG is enabled.

Screenshot 2023-08-04 at 11 05 48 AM
Screenshot 2023-08-04 at 11 06 09 AM
Screenshot 2023-08-04 at 11 06 17 AM
Screenshot 2023-08-04 at 11 06 39 AM
Screenshot 2023-08-04 at 11 06 46 AM
Screenshot 2023-08-04 at 11 06 54 AM

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51119)

### CODE REVIEW

**Comment by GoelPratyush99** [comment]
trigger uts

**Comment by GoelPratyush99** [comment]
trigger uts

**Comment by GoelPratyush99** [comment]
trigger ut3

**Comment by GoelPratyush99** [comment]
trigger uts

**Comment by GoelPratyush99** [comment]
trigger uts

**Comment by GoelPratyush99** [comment]
trigger ut2

**Comment by GoelPratyush99** [comment]
trigger ut7

**Comment by GoelPratyush99** [comment]
trigger ut7

**Review by abhinavsingh1196 — CHANGES_REQUESTED** [review]
> Reviewed 6 of 16 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51119)* status: Some of the checks are too old. 10 file(s) to review. 2 change requests. 0 of 2 approvals obtained. 

___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/k8s/K8sCanaryRequestHandler.java` line 295 at r3](https://reviewable.io/reviews/harness/harness-core/51119#-NbFc-Ob78LL1iWMXK1C:-NbFc-Ob78LL1iWMXK1D:b-ptbt7w) ([raw file](https://github.com/harness/harness-core/blob/2052674ca9531f0bcb2b61e2b40e3c55981e43a3/930-delegate-tasks/src/main/java/io/harness/delegate/k8s/K8sCanaryRequestHandler.java#L295)):*
> ```Java
> 
>     k8sCanaryBaseHandler.updateDestinationRuleManifestFilesWithSubsets(
>         resources, request.isDisableFabric8() ? null : k8sCanaryHandlerConfig.getKubernetesConfig(), logCallback);
> ```

not a fan of this approach. When we GA the flag, then this will either be replaced by null or require another minor refactor.
we should probably just do method overload (with and without KubernetesConfig based on the fabric8 FF value). 

This will also break down downstream methods and avoid internal checks like 
`if (kubernetesConfig == null)`
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/k8s/K8sDryRunManifestRequestHandler.java` line 154 at r3](https://reviewable.io/reviews/harness/harness-core/51119#-NbFcljY8AKdp66PxO6O:-NbFcljY8AKdp66PxO6P:b-hewrfu) ([raw file](https://github.com/harness/harness-core/blob/2052674ca9531f0bcb2b61e2b40e3c55981e43a3/930-delegate-tasks/src/main/java/io/harness/delegate/k8s/K8sDryRunManifestRequestHandler.java#L154)):*
> ```Java
>         manifestOverrideFiles, this.kubernetesConfig, this.manifestFilesDirectory, this.releaseName,
>         request.isLocalOverrideFeatureFlag(), isErrorFrameworkSupported(), request.isInCanaryWorkflow(),
>         request.isDisableFabric8(), executionLogCallback);
> ```

does this need a further refactor? method params have become too many now. `request` var is available to the method already, we can get all these booleans from it directly? 
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/k8s/rolling/K8sRollingBaseHandler.java` line 253 at r3](https://reviewable.io/reviews/harness/harness-core/51119#-NbFdJDK5NAJ-0naS4_U:-NbFdJDK5NAJ-0naS4_V:b-al2tnh) ([raw file](https://github.com/harness/harness-core/blob/2052674ca9531f0bcb2b61e2b40e3c55981e43a3/930-delegate-tasks/src/main/java/io/harness/delegate/k8s/rolling/K8sRollingBaseHandler.java#L253)):*
> ```Java
> 
>     if (isInCanaryWorkflow) {
>       updateDestinationRuleWithSubsets(executionLogCallback, resources, shouldDisableFabric8 ? null : kubernetesConfig);
> ```

same as above, we could put another condition here based on shouldDisableFabric8 and call different methods based on it
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/k8s/rolling/K8sRollingRequestHandler.java` line 331 at r3](https://reviewable.io/reviews/harness/harness-core/51119#-NbFdafN7SQ-A-_Yh2sP:-NbFdafN7SQ-A-_Yh2sQ:b2aw0ir) ([raw file](https://github.com/harness/harness-core/blob/2052674ca9531f0bcb2b61e2b40e3c55981e43a3/930-delegate-tasks/src/main/java/io/harness/delegate/k8s/rolling/K8sRollingRequestHandler.java#L331)):*
> ```Java
>         manifestOverrideFiles, this.kubernetesConfig, this.manifestFilesDirectory, this.releaseName,
>         request.isLocalOverrideFeatureFlag(), isErrorFrameworkSupported(), request.isInCanaryWorkflow(),
>         request.isDisableFabric8(), executionLogCallback);
> ```

same as above, request is already available. We can avoid passing a few params

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/istio/IstioTaskHelper.java [hunk 1]
```diff
@@ -189,7 +202,85 @@ public void updateVirtualServiceWithDestinationWeights(List<IstioDestinationWeig
...
   }
 
-  private VirtualService updateVirtualServiceManifestFilesWithRoutes(List<KubernetesResource> resources,
+  private void validateRoutesInVirtualService(KubernetesResource virtualService) {
+    List<Object> http = (List<Object>) virtualService.getField("spec.http");
+    List<Object> tcp = (List<Object>) virtualService.getField("spec.tcp");
+    List<Object> tls = (List<Object>) virtualService.getField("spec.tls");
+
+    if (isEmpty(http)) {
+      throw new InvalidRequestException(
```
Comment (abosii-harness) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/istio/IstioTaskHelper.java:211]: Isn't this a breaking change? Did we used to throw exceptions when something is not present or not existing?

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/istio/IstioTaskHelper.java [hunk 2]
```diff
@@ -255,17 +350,21 @@ public DestinationRule updateDestinationRuleManifestFilesWithSubsets(List<Kubern
       throw new InvalidRequestException(msg, USER);
     }
 
+    KubernetesResource kubernetesResource = destinationRuleResources.get(0);
+    if (kubernetesConfig == null) {
```
Comment (abosii-harness) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/istio/IstioTaskHelper.java:354]: It's very confusing this logic, either refactor or it or add some comments

**Review by abosii-harness — CHANGES_REQUESTED** [review]

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/istio/IstioTaskHelper.java (hunk 1 above)
Comment (GoelPratyush99) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/istio/IstioTaskHelper.java:211]: Not a breaking change as we used to this validation earlier as well `io.harness.delegate.task.k8s.istio.IstioTaskHelper#validateRoutesInVirtualService(io.fabric8.istio.api.networking.v1alpha3.VirtualService)`

**Review by GoelPratyush99 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51119)* status: Some of the checks are too old. 10 file(s) to review. 3 change requests. 5. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/k8s/K8sCanaryRequestHandler.java` line 295 at r3](https://reviewable.io/reviews/harness/harness-core/51119#-NbFc-Ob78LL1iWMXK1C:-NbIL_1E9rnpp9OHgmb2:bqedoqu) ([raw file](https://github.com/harness/harness-core/blob/2052674ca9531f0bcb2b61e2b40e3c55981e43a3/930-delegate-tasks/src/main/java/io/harness/delegate/k8s/K8sCanaryRequestHandler.java#L295)):*
<details><summary><i>Previously, abhinavsingh1196 (Abhinav Singh) wrote…</i></summary><blockquote>

not a fan of this approach. When we GA the flag, then this will either be replaced by null or require another minor refactor.
we should probably just do method overload (with and without KubernetesConfig based on the fabric8 FF value). 

This will also break down downstream methods and avoid internal checks like 
`if (kubernetesConfig == null)`
</blockquote></details>

I will be refactoring once we GA this FF. Personally I would pick this way as it much easier to do minor refactoring in the end rather than add a whole set of new methods with and without kubeconfig field and then remove all these additional classes.
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/k8s/K8sDryRunManifestRequestHandler.java` line 154 at r3](https://reviewable.io/reviews/harness/harness-core/51119#-NbFcljY8AKdp66PxO6O:-NbIMJ2E6hm7NugAMFrm:bb7zuxi) ([raw file](https://github.com/harness/harness-core/blob/2052674ca9531f0bcb2b61e2b40e3c55981e43a3/930-delegate-tasks/src/main/java/io/harness/delegate/k8s/K8sDryRunManifestRequestHandler.java#L154)):*
<details><summary><i>Previously, abhinavsingh1196 (Abhinav Singh) wrote…</i></summary><blockquote>

does this need a further refactor? method params have become too many now. `request` var is available to the method already, we can get all these booleans from it directly? 
</blockquote></details>

This request is being passed to K8sRollingBaseHandler.java. We would need to type cast the K8sDeployRequest to K8sDryRunManifestRequest and K8sRollingDeployRequest to get the value of all the booleans.
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/istio/IstioTaskHelper.java` line 75 at r3](https://reviewable.io/reviews/harness/harness-core/51119#-NbFEI7x8qTbI2v_S1cM:-NbIOcW8CLaGa4l6WHxF:b-ptxi4k) ([raw file](https://github.com/harness/harness-core/blob/2052674ca9531f0bcb2b61e2b40e3c55981e43a3/930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/istio/IstioTaskHelper.java#L75)):*
<details><summary><i>Previously, alebuh (Aleksa Buha) wrote…</i></summary><blockquote>

is there a reason why we are returning Object type instead of  List<io.harness.k8s.model.istio.Subset> ?
</blockquote></details>

Changed it
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/istio/IstioTaskHelper.java` line 90 at r3](https://reviewable.io/reviews/harness/harness-core/51119#-NbFCvkLEJcyNdsE-P-G:-NbIOgYrBsPcWKUsmX4m:biwdm18) ([raw file](https://github.com/harness/harness-core/blob/2052674ca9531f0bcb2b61e2b40e3c55981e43a3/930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/istio/IstioTaskHelper.java#L90)):*
<details><summary><i>Previously, alebuh (Aleksa Buha) wrote…</i></summary><blockquote>

Might be better to use switch here instead of 4 if 
</blockquote></details>

We are not using HarnessLabelValues.colorBlue and HarnessLabelValues.colorGreen anywhere so I have removed it :)

**Comment by GoelPratyush99** [comment]
trigger uts

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/istio/IstioTaskHelper.java (hunk 2 above)
Comment (GoelPratyush99) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/istio/IstioTaskHelper.java:354]: Added comments

**Comment by GoelPratyush99** [comment]
trigger uts

**Comment by GoelPratyush99** [comment]
trigger copyrightcheck

**Comment by GoelPratyush99** [comment]
trigger ut7

**Comment by GoelPratyush99** [comment]
trigger uts

**Review by alebuh — CHANGES_REQUESTED** [review]
> Reviewed 1 of 2 files at r4, 1 of 1 files at r5, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51119)* status: 11 file(s) to review. 3 change requests. 4. unresolved discussions. 0 of 2 approvals obtained. Waiting on team(s): k8s-team. 

___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/k8s/K8sCanaryRequestHandler.java` line 295 at r3](https://reviewable.io/reviews/harness/harness-core/51119#-NbFc-Ob78LL1iWMXK1C:-NbJhPMjAMxNAIvQZtXG:bssszgw) ([raw file](https://github.com/harness/harness-core/blob/2052674ca9531f0bcb2b61e2b40e3c55981e43a3/930-delegate-tasks/src/main/java/io/harness/delegate/k8s/K8sCanaryRequestHandler.java#L295)):*
<details><summary><i>Previously, GoelPratyush99 (Pratyush Goel) wrote…</i></summary><blockquote>

I will be refactoring once we GA this FF. Personally I would pick this way as it much easier to do minor refactoring in the end rather than add a whole set of new methods with and without kubeconfig field and then remove all these additional classes.
</blockquote></details>

I agree with @abhinavsingh1196 on this. It would be better to use overloading. When we GA this FF it will require some code refactoring if we leave it like this. I think it's better to refactor this now a little bit more so that when we GA it and remove FF from the code we don't need to change the logic of the code.
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/istio/IstioTaskHelper.java` line 90 at r3](https://reviewable.io/reviews/harness/harness-core/51119#-NbFCvkLEJcyNdsE-P-G:-NbJimV97e9LV_nEqgQL:b-sfzc1u) ([raw file](https://github.com/harness/harness-core/blob/2052674ca9531f0bcb2b61e2b40e3c55981e43a3/930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/istio/IstioTaskHelper.java#L90)):*
<details><summary><i>Previously, GoelPratyush99 (Pratyush Goel) wrote…</i></summary><blockquote>

We are not using HarnessLabelValues.colorBlue and HarnessLabelValues.colorGreen anywhere so I have removed it :)
</blockquote></details>

That's fine, but if we are leaving like this, it would be better to use else if. No need to change subsateName again if it's trackCanary right?

**Review by alebuh — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51119)* status: 11 file(s) to review. 3 change requests. 4. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/istio/IstioTaskHelper.java` line 90 at r3](https://reviewable.io/reviews/harness/harness-core/51119#-NbFCvkLEJcyNdsE-P-G:-NbJjfrG6lh0PMLlUcel:brpu3z1) ([raw file](https://github.com/harness/harness-core/blob/2052674ca9531f0bcb2b61e2b40e3c55981e43a3/930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/istio/IstioTaskHelper.java#L90)):*
<details><summary><i>Previously, alebuh (Aleksa Buha) wrote…</i></summary><blockquote>

That's fine, but if we are leaving like this, it would be better to use else if. No need to change subsateName again if it's trackCanary right?
</blockquote></details>

check subsateName*

**Review by GoelPratyush99 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51119)* status: 11 file(s) to review. 3 change requests. 4. unresolved discussions. 0 of 2 approvals obtained. Waiting on team(s): k8s-team. 

___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/k8s/K8sCanaryRequestHandler.java` line 295 at r3](https://reviewable.io/reviews/harness/harness-core/51119#-NbFc-Ob78LL1iWMXK1C:-NbKAgqH1imAAVEjlPFY:bldceu1) ([raw file](https://github.com/harness/harness-core/blob/2052674ca9531f0bcb2b61e2b40e3c55981e43a3/930-delegate-tasks/src/main/java/io/harness/delegate/k8s/K8sCanaryRequestHandler.java#L295)):*
<details><summary><i>Previously, alebuh (Aleksa Buha) wrote…</i></summary><blockquote>

I agree with @abhinavsingh1196 on this. It would be better to use overloading. When we GA this FF it will require some code refactoring if we leave it like this. I think it's better to refactor this now a little bit more so that when we GA it and remove FF from the code we don't need to change the logic of the code.
</blockquote></details>

I would need to remove the code later as the main task is to remove fabric8 library that we are using currently

**Review by GoelPratyush99 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51119)* status: 11 file(s) to review. 3 change requests. 5. unresolved discussions. 0 of 2 approvals obtained. Waiting on team(s): k8s-team. 

___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/istio/IstioTaskHelper.java` line 90 at r3](https://reviewable.io/reviews/harness/harness-core/51119#-NbFCvkLEJcyNdsE-P-G:-NbKBR6m3V763QHuW_bW:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/2052674ca9531f0bcb2b61e2b40e3c55981e43a3/930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/istio/IstioTaskHelper.java#L90)):*
<details><summary><i>Previously, alebuh (Aleksa Buha) wrote…</i></summary><blockquote>

check subsateName*
</blockquote></details>

Done

**Comment by GoelPratyush99** [comment]
trigger uts

**Comment by GoelPratyush99** [comment]
trigger uts

**Review by abosii-harness — CHANGES_REQUESTED** [review]

**Review by alebuh — DISMISSED** [review]
> Reviewed 1 of 1 files at r7.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51119)* status: 11 file(s) to review. 2 change requests. 6. unresolved discussions. 1 of 2 approvals obtained.

File: 970-api-services-beans/src/main/java/io/harness/k8s/model/istio/HttpRouteDestination.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonRootName("subset")
```
[the remark this answers is not in the document]
Comment (GoelPratyush99) [reply 970-api-services-beans/src/main/java/io/harness/k8s/model/istio/HttpRouteDestination.java:26]: removed it

**Comment by GoelPratyush99** [comment]
trigger uts

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/istio/IstioTaskHelper.java [hunk 4]
```diff
@@ -71,31 +72,36 @@ public class IstioTaskHelper {
...
+      subset.setLabels(getLabelsFromSubsetName(subsetName));
+      subsets.add(subset);
+    }
+
+    return subsets;
+  }
+
+  private Map<String, String> getLabelsFromSubsetName(String subsetName) {
+    Map<String, String> labels = new HashMap<>();
+    if (subsetName.equals(HarnessLabelValues.trackCanary)) {
```
Comment (abosii-harness) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/istio/IstioTaskHelper.java:90]: What about the color labels?

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/istio/IstioTaskHelper.java [hunk 5]
```diff
@@ -189,7 +195,85 @@ public void updateVirtualServiceWithDestinationWeights(List<IstioDestinationWeig
...
+    return (String) virtualService.getField("spec.http[0].route[0].destination.host");
+  }
+
+  private List<HttpRouteDestination> generateDestinationWeights(
+      List<IstioDestinationWeight> istioDestinationWeights, String host, String portNumber) throws IOException {
+    List<HttpRouteDestination> destinationWeights = new ArrayList<>();
+
+    for (IstioDestinationWeight istioDestinationWeight : istioDestinationWeights) {
+      String destinationYaml = getDestinationYaml(istioDestinationWeight.getDestination(), host);
+      io.harness.k8s.model.istio.Destination destination =
```
Comment (abosii-harness) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/istio/IstioTaskHelper.java:239]: nit: Can we reverse it with fabric8 package? So we will import our package but reference by package path the fabric8?

**Review by abosii-harness — CHANGES_REQUESTED** [review]

**Comment by GoelPratyush99** [comment]
trigger uts

**Comment by GoelPratyush99** [comment]
trigger ut3

**Review by alebuh — APPROVED** [review]
> Reviewed 5 of 5 files at r8, 2 of 2 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51119)* status: 11 file(s) to review. 2 change requests. 7. unresolved discussions. 1 of 2 approvals obtained.

**Review by abosii-harness — APPROVED** [review]

**Comment by GoelPratyush99** [comment]
trigger ut4

**Review by TarunUba1 — APPROVED** [review]
Add UTs also for RollingRequestHandler and Base Handler
Update:
-ignore as the main changes are already tested  for. IstioTaskHelper

**Comment by GoelPratyush99** [comment]
trigger review

File: 970-api-services-beans/src/main/java/io/harness/k8s/model/istio/Destination.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@CodePulse(module = ProductModule.CDS, unitCoverageRequired = true, components = {HarnessModuleComponent.CDS_K8S})
+public class Destination {
```
Comment (markoharness) [inline 970-api-services-beans/src/main/java/io/harness/k8s/model/istio/Destination.java:24]: Should we have OwnedBy annotation?

File: 970-api-services-beans/src/main/java/io/harness/k8s/model/istio/HttpRouteDestination.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@CodePulse(module = ProductModule.CDS, unitCoverageRequired = true, components = {HarnessModuleComponent.CDS_K8S})
+public class HttpRouteDestination {
```
Comment (markoharness) [inline 970-api-services-beans/src/main/java/io/harness/k8s/model/istio/HttpRouteDestination.java:24]: Should we have OwnedBy annotation?

File: 970-api-services-beans/src/main/java/io/harness/k8s/model/istio/PortSelector.java [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@CodePulse(module = ProductModule.CDS, unitCoverageRequired = true, components = {HarnessModuleComponent.CDS_K8S})
+public class PortSelector {
```
Comment (markoharness) [inline 970-api-services-beans/src/main/java/io/harness/k8s/model/istio/PortSelector.java:24]: Should we have OwnedBy annotation?

File: 970-api-services-beans/src/main/java/io/harness/k8s/model/istio/Subset.java [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@CodePulse(module = ProductModule.CDS, unitCoverageRequired = true, components = {HarnessModuleComponent.CDS_K8S})
+public class Subset {
```
Comment (markoharness) [inline 970-api-services-beans/src/main/java/io/harness/k8s/model/istio/Subset.java:25]: Should we have OwnedBy annotation?

**Review by markoharness — APPROVED** [review]

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/istio/IstioTaskHelper.java (hunk 4 above)
Comment (GoelPratyush99) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/istio/IstioTaskHelper.java]: I cannot find any usage for it. Still added this code back for precaution.

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/istio/IstioTaskHelper.java (hunk 5 above)
Comment (GoelPratyush99) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/istio/IstioTaskHelper.java]: Done

File: 970-api-services-beans/src/main/java/io/harness/k8s/model/istio/Destination.java (hunk 6 above)
Comment (GoelPratyush99) [reply 970-api-services-beans/src/main/java/io/harness/k8s/model/istio/Destination.java:24]: Shouldn't code pulse be enough?

File: 970-api-services-beans/src/main/java/io/harness/k8s/model/istio/HttpRouteDestination.java (hunk 7 above)
Comment (GoelPratyush99) [reply 970-api-services-beans/src/main/java/io/harness/k8s/model/istio/HttpRouteDestination.java:24]: Shouldn't code pulse be enough?

File: 970-api-services-beans/src/main/java/io/harness/k8s/model/istio/PortSelector.java (hunk 8 above)
Comment (GoelPratyush99) [reply 970-api-services-beans/src/main/java/io/harness/k8s/model/istio/PortSelector.java:24]: Shouldn't code pulse be enough?

File: 970-api-services-beans/src/main/java/io/harness/k8s/model/istio/Subset.java (hunk 9 above)
Comment (GoelPratyush99) [reply 970-api-services-beans/src/main/java/io/harness/k8s/model/istio/Subset.java:25]: Shouldn't code pulse be enough?

**Review by alebuh — APPROVED** [review]

**Review by abhinavsingh1196 — APPROVED** [review]
> Reviewed 2 of 16 files at r1, 1 of 3 files at r2, 1 of 1 files at r5, 5 of 5 files at r8, 2 of 2 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51119)* status: 8. unresolved discussions. 

___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/k8s/K8sDryRunManifestRequestHandler.java` line 154 at r3](https://reviewable.io/reviews/harness/harness-core/51119#-NbFcljY8AKdp66PxO6O:-NbUbv42034PsKWYL90A:b-gmxw70) ([raw file](https://github.com/harness/harness-core/blob/2052674ca9531f0bcb2b61e2b40e3c55981e43a3/930-delegate-tasks/src/main/java/io/harness/delegate/k8s/K8sDryRunManifestRequestHandler.java#L154)):*
<details><summary><i>Previously, GoelPratyush99 (Pratyush Goel) wrote…</i></summary><blockquote>

This request is being passed to K8sRollingBaseHandler.java. We would need to type cast the K8sDeployRequest to K8sDryRunManifestRequest and K8sRollingDeployRequest to get the value of all the booleans.
</blockquote></details>

Discussed offline, further typecasting could possibly cause problems later. Keeping it as is for now to avoid causing new issues.

**Pull request merged**